### PR TITLE
GiftWrapping changes on proposed schema

### PR DESCRIPTION
**Pull request by avattam06**
## Problem

Some changes on the Gift wrapping schema are added which are based on https://github.com/magento/architecture/blob/master/design-documents/graph-ql/coverage/gift-wrapping.md. Submitting the changes for review as part of this PR https://github.com/magento/partners-magento2ee/pull/246.

## Solution

Submitting the new changes for review.

## Requested Reviewers

@paliarush 
@prabhuram93

### CODE REVIEW

**Comment by avattam06** [comment]
@paliarush, On line 201 for `setGiftOptionsOnCart` mutation name, Can we change it to `setGiftOptionsOnAllCart` or `setGiftOptionsOnWholeCart` as `setgiftOptionsonCart` will add the provided giftwrapping ID on the whole cart but not on the item level.

 Because we are using `updateCartItems` to add gift wrapping on to the cart items.

 Is it better to cover adding `giftWrapping` on item level and on order level on the same mutation `setGiftOptionsOnAllCart`

Please suggest your thoughts on this?

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 1]
```diff
@@ -56,25 +61,40 @@ type CartPrices {
...
-type GiftWrapping {
-    id: ID! @doc(description: "Gift wrapping unique identifier")
-    design: String! @doc(description: "Gift wrapping design name")
-    price: Money! @doc(description: "Gift wrapping price")
-    image: GiftWrappingImage! @doc(description: "Gift wrapping preview image")
+###### StoreConfig ######
+type StoreConfig {
+    allow_gift_wrapping_on_order: String @doc(description: "Allow Gift Wrapping on Order Level")
+    allow_gift_wrapping_on_order_items: String @doc(description: "Allow Gift Wrapping for Order Items")
+    allow_gift_receipt: String @doc(description: "Allow Gift Receipt")
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md:71]: We need to replace `Configuration` section (which describes the same but in xml format) in this document with this new declaration.

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 2]
```diff
@@ -56,25 +61,40 @@ type CartPrices {
     gift_options: GiftOptionsPrices @doc(description: "The list of prices for the selected gift options")
 }
 
-type SalesTotalsInterface {
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md:59]: Why is this type extension removed?

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 3]
```diff
@@ -45,6 +45,11 @@ type SalesItemInterface {
     gift_message: GiftMessage @doc(description: "The entered gift message for the order item")
 }
 
+interface OrderItemInterface {
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md:48]: Why do we have both now `SalesItemInterface` and `OrderItemInterface`? Should `SalesItemInterface` just be renamed to `OrderItemInterface`?

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 1 above)
Comment (avattam06) [reply design-documents/graph-ql/coverage/gift-wrapping.md:71]: Okay, Those changes are already added to this PR magento/partners-magento2ee#246. I will change them here too.

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 2 above)
Comment (avattam06) [reply design-documents/graph-ql/coverage/gift-wrapping.md:59]: It was removed as part of the customer orders schema.

File: design-documents/graph-ql/coverage/gift-wrapping.md (hunk 3 above)
Comment (avattam06) [reply design-documents/graph-ql/coverage/gift-wrapping.md:48]: I was renamed as part of customer orders schema changes.

**Comment by avattam06** [comment]
> @paliarush, On line 201 for `setGiftOptionsOnCart` mutation name, Can we change it to `setGiftOptionsOnAllCart` or `setGiftOptionsOnWholeCart` as `setgiftOptionsonCart` will add the provided giftwrapping ID on the whole cart but not on the item level.
> 
> Because we are using `updateCartItems` to add gift wrapping on to the cart items.
> 
> Is it better to cover adding `giftWrapping` on item level and on order level on the same mutation `setGiftOptionsOnAllCart`
> 
> Please suggest your thoughts on this?

 Giftwrapping Schema @paliarush

**Comment by avattam06** [comment]
@nrkapoor, Added new schema changes for the gift-wrapping on whole cart and item level. Please check.

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 4]
```diff
@@ -176,17 +196,27 @@ type CartItemUpdateInput {
     gift_wrapping_id: ID @doc(description: "The unique identifier of the gift wrapping to be used for the cart item")
     gift_message: GiftMessageInput @doc(description: "Gift message details for the cart item")
 }
+type RemoveItemFromCartInput {
```
Comment (DrewML) [inline design-documents/graph-ql/coverage/gift-wrapping.md:199]: Just needs to switch from `type` to `input`

Comment (avattam06) [reply design-documents/graph-ql/coverage/gift-wrapping.md:199]: yes, you are right. I will change that.

**Comment by paliarush** [comment]
@avattam06 regarding https://github.com/magento/architecture/pull/416#issuecomment-670090794 I don't see a strong reason to change the name of `setGiftOptionsOnCart`. The two proposed names look strange to me. We can clarify the effect of this mutation via docs.

File: design-documents/graph-ql/coverage/gift-wrapping.md [hunk 5]
```diff
@@ -176,17 +196,27 @@ type CartItemUpdateInput {
...
+    gift_wrapping_id: ID @doc(description: "The unique identifier of the gift wrapping to be used for removing from the cart item")
+}
 
 type Mutation {
     setGiftOptionsOnCart(cart_id: String!, gift_message: GiftMessageInput, gift_wrapping_id: ID, gift_receipt_included: Boolean, printed_card_included: Boolean): SetGiftOptionsOnCartOutput @doc(description: "Set gift options like gift wrapping or gift message for the entire cart")
+    removeGiftOptionsOnCart(cart_id: String!, gift_wrapping_id: ID, gift_receipt_included: Boolean, printed_card_included: Boolean): RemoveGiftOptionsOnCartOutput @doc(description: "Remove gift options gift wrapping for the entire cart")
```
Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md:205]:
```suggestion
    removeGiftOptionsFromCart(cart_id: String!, gift_wrapping_id: ID, gift_receipt_included: Boolean, printed_card_included: Boolean): RemoveGiftOptionsOnCartOutput @doc(description: "Remove gift options gift wrapping for the entire cart")
```

Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md:205]: Based on recent agreement we must always use `input` wrapper for mutation arguments.

Comment (paliarush) [inline design-documents/graph-ql/coverage/gift-wrapping.md:205]: Why do we need separate mutation? All gift options can be unset/changed using `setGiftOptionsOnCart`

Comment (avattam06) [reply design-documents/graph-ql/coverage/gift-wrapping.md:205]: I don't think we are doing it now, we are only able to set the gift options. I don't see the unset even for merged giftmessage features.

**Comment by paliarush** [comment]
@avattam06 the merge is blocked because of the conflicts.

**Pull request merged**